src/smart-components/group/service-account/add-group-service-accounts.tsx

```diff
@@ -1,14 +1,171 @@
-import React from 'react';
+import React, { Fragment, useEffect, useState } from 'react';
 import { useIntl } from 'react-intl';
-import { Modal, ModalVariant } from '@patternfly/react-core';
+import { useDispatch, useSelector } from 'react-redux';
+import { Alert, Button, Modal, ModalVariant, Stack, StackItem, TextContent } from '@patternfly/react-core';
+import { useParams } from 'react-router-dom';
+import DateFormat from '@redhat-cloud-services/frontend-components/DateFormat';
+import useChrome from '@redhat-cloud-services/frontend-components/useChrome';
+import AppLink from '../../../presentational-components/shared/AppLink';
+import useAppNavigate from '../../../hooks/useAppNavigate';
+import { TableToolbarView } from '../../../presentational-components/shared/table-toolbar-view';
+import { fetchServiceAccounts } from '../../../redux/actions/service-account-actions';
+import { ServiceAccountsState } from '../../../redux/reducers/service-account-reducer';
+import { getDateFormat } from '../../../helpers/shared/helpers';
+import { RowProps } from '../../user/user-table-helpers';
+import { LAST_PAGE, ServiceAccount } from '../../../helpers/service-account/service-account-helper';
+import { DEFAULT_ACCESS_GROUP_ID } from '../../../utilities/constants';
+import { addServiceAccountsToGroup } from '../../../redux/actions/group-actions';
 import messages from '../../../Messages';
+import './group-service-accounts.scss';
 
-const AddGroupServiceAccounts: React.FC = () => {
+interface AddGroupServiceAccountsProps {
+  submitRoute: string;
+  cancelRoute: string;
+}
+
+export interface PaginationProps {
+  count?: number;
+  limit: number;
+  offset: number;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const reducer = ({ serviceAccountReducer, groupReducer: { systemGroup } }: { serviceAccountReducer: ServiceAccountsState; groupReducer: any }) => ({
```

Comment:
We know the shape of the `groupReducer`
```suggestion
const reducer = ({ serviceAccountReducer, groupReducer: { systemGroup } }: { serviceAccountReducer: ServiceAccountsState; groupReducer: { systemGroup?: { uuid: string } } }) => ({
```